pyrallis/utils.py

```diff
@@ -572,6 +574,60 @@ def deflatten(d: Dict[str, Any], sep: str = '.'):
     return deflat_d
 
 
+def parse_string(s):
+    if Config.get_config_type() == ConfigType.YAML:
+        import yaml
+        return yaml.safe_load(s)
+    elif Config.get_config_type() == ConfigType.JSON:
+        import json
+        try:
+            return json.loads(s)
+        except json.decoder.JSONDecodeError:
+            return s # No parsing to be done, simply return value
```

Comment:
This means that `parse_string('{"val": 7')` will not raise. Is this desired?

Reply:
Not really, though I wonder how to better tackle that.
The thing is that I'm trying to use the parsing package (`yaml`/`json`/`toml`) for the first stage of processing the cmd strings (In order to ensure consistency between arguments loaded from cmd and files). Now, in yaml the `yaml.safe_load` function is able to handle both strings and collections.
```python
yaml.safe_load('A string') # 'A string'
```

```python
yaml.safe_load('[1,2,3]') # [1,2,3]
```

```python
yaml.safe_load('[1,2,3') # Error
```

While the `json.loads` function can successfully load collections, it raises an Exception for strings that it cannot parse into a known collection, so not sure how to differentiate between string values and other decoding errors

Reply:
You are right, and this is a good protection to start with. I think maybe it is possible to come up with a mechanism to determine different kinds of strings (e.g. do all 'collection' strings begin and end with one of `()`/`[]`/`{}`? This can be checked) and then you have somethin similar to `safe` loading. Personally though I won't bother. And also it can be fragile. Instead I think maybe a warning can help? In the cases where the string is treated as a "raw" string...

Reply:
Good idea, will add a warning

Reply:
On a second thought, warnings can really annoying here, because I'll get a warning for every string value I pass. I think that the parsingError that will be raised is good enough for now